client/state/data-layer/wpcom/sites/activity/test/index.js

```diff
@@ -80,13 +77,11 @@ describe( 'receiveActivityLog', () => {
 describe( 'receiveActivityLogError', () => {
 	it( 'should dispatch activity log error action', () => {
 		const dispatch = sinon.spy();
-		receiveActivityLogError( { dispatch }, { siteId: SITE_ID }, ERROR_RESPONSE );
-		expect( dispatch ).to.have.been.calledWith(
-			activityLogError( SITE_ID, {
-				error: 'unknown_blog',
-				message: 'Unknown blog',
-			} )
-		);
+		receiveActivityLogError( { dispatch } );
+		expect( dispatch ).to.have.been.called.once;
+		expect( dispatch ).to.have.been.calledWithMatch( {
+			type: errorNotice( translate( 'Error receiving activity for site.' ) ).type,
```

Comment:
actually @eliorivero I found what I was looking for. this is hard because `errorNotice()` is a stateful action creator (a great reason to make pure action creators). it depends on `uniqueId()` to return a new id on each call.

we can get around this nonsense in one of two ways:
 - take advantage of the fact that we know what's happening inside the implementation of `errorNotice()` and pass an id override to our expected call -> `errorNotice( 'Thanks', { id: 1 } )`
 - do that to a lesser extend and mock `uniqueId()` for this test

A while back I created [impure lodash](https://github.com/Automattic/wp-calypso/tree/925fce4902a758265af12818fc6484a41b14311b/client/lib/impure-lodash) as a way to work around these nasty testing bits. If we update [createNotice()](https://github.com/Automattic/wp-calypso/blob/925fce4902a758265af12818fc6484a41b14311b/client/state/notices/actions.js#L7) to import `impure-lodash` instead of `lodash` then we can mock it out and supply our own test ids (never use `0` or `1` in tests!) in the tests. Although we're still encoding implementation details into the tests, we're at least doing so from outside of the unit under test.

For an example see the [MediaUtils tests](https://github.com/Automattic/wp-calypso/blob/20c90e4e7a397c19b66133a8e5df00fdb046b4c8/client/lib/media/test/utils.js#L18) brought in in #15096

---

In other words, how about update `createNotice()` to use `impure-lodash`, then mock it in this test and do what we wanted to do from the beginning?

```js
expect( dispatch ).to.have.been.calledWith( errorNotice( 'Error receiving…' ) )
```

Reply:
I've updated this to pass `{ id: '1' }` as parameter. I decided to do it this way because:
1. I didn't want to unnecessarily mock this with impure-lodash since it's something that should be replaced eventually. Thanks for mentioning though, when it's impossible to do it in any other way, I'll make use of it.
2. `errorNotice` allowed to pass the id as a parameter so I'm not short-circuiting the test

Reply:
how about we use a non-`1` id? they are notoriously troublesome for tests because on account of our assumptions tests with things that start and continue tend to start from `0` or `1`. our test might pass when we give it a `1` but then fail when we legitimately give it a `13` :wink:

Reply:
It's just that `receiveActivityLogError` calls `errorNotice` with no `id` argument so that will be `'1'`.

Reply:
oh snap! this is so awkward! alright, ship it or maybe update `createNotice()` - I'll try and get a PR ready for that anyway and I'll let you know so you can update these tests - should be like a one or two line PR

Reply:
I'll ship it since `createNotice` is out of the scope of this PR. Do let me know if you create that PR and I'll update this. Thanks for the review 👍

Reply:
No worries, though did you see #18849 where I did the change?

Reply:
Saw it now, approved.